### PR TITLE
[Feature]Support `Show databases from catalog;` for dla connector

### PR DESCRIPTION
**Pull request by adzfolc**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 9 / 24 (37.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|13|07.69%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|4|50.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|6|7|85.71%|

File: fe/fe-core/src/main/java/com/starrocks/analysis/ShowDbStmt.java
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/analysis/ShowDbStmt.java:42]: I think this should be catalog? Do you want to show databases from specific catalog?

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:540]: why catch NullPointerException? I think it is not common to catch NullPointer.

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:44]: why not delete showDbFromCatalogStatement?

Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:44]: > why not delete showDbFromCatalogStatement?

need to wait for https://github.com/StarRocks/starrocks/pull/6045 got merged

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:540]: > why catch NullPointerException? I think it is not common to catch NullPointer.

If execute dql `show databases from catalog_a`,while catalog_a doesn't exist, it would throw NullPointer here.

File: fe/fe-core/src/main/java/com/starrocks/analysis/ShowDbStmt.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/analysis/ShowDbStmt.java:42]: > I think this should be catalog? Do you want to show databases from specific catalog?

Yes.

output is like blow

MySQL [(none)]> CREATE EXTERNAL CATALOG hive_catalog_1 COMMENT "hive_catalog" PROPERTIES("type"="hive", "hive.metastore.uris"="thrift://127.0.0.1:9083");
Query OK, 0 rows affected (0.00 sec)

MySQL [(none)]> show catalogs;
+----------------+------+--------------+
| Catalog        | Type | Comment      |
+----------------+------+--------------+
| hive_catalog_1 | hive | hive_catalog |
+----------------+------+--------------+
1 row in set (0.00 sec)
MySQL [(none)]> CREATE EXTERNAL CATALOG hive_catalog_1 COMMENT "hive_catalog" PROPERTIES("type"="hive", "hive.metastore.uris"="thrift://127.0.0.1:9083");
Query OK, 0 rows affected (0.00 sec)

MySQL [(none)]> show catalogs;
+----------------+------+--------------+
| Catalog        | Type | Comment      |
+----------------+------+--------------+
| hive_catalog_1 | hive | hive_catalog |
+----------------+------+--------------+
1 row in set (0.00 sec)

MySQL [(none)]> show databases from hive_catalog_1;
ERROR 1064 (HY000): Failed to get all databases name on thrift://127.0.0.1:9083
**hms for hive_catalog_1 is down**

MySQL [(none)]> show databases from hive_catalog_1;
+----------+
| Database |
+----------+
| default  |
| tn_test  |
| un_test  |
+----------+
3 rows in set (1.10 sec)

MySQL [(none)]> show databases from hive_catalog_1;
+----------+
| Database |
+----------+
| default  |
| tn_test  |
| un_test  |
+----------+
3 rows in set (1.10 sec)

MySQL [(none)]> show databases from hive_catalog_2;
ERROR 1064 (HY000): Can not acquire databases from hive_catalog_2, please check catalog config!
**hive_catalog_2 is not configured**

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 8 / 25 (32.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|15|06.67%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|4|50.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|5|6|83.33%|

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:542]: why throw nullptr exception. I think a normal exception should be thrown. Such as AnalysisException.

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1371]: why mush add cast string here?

Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java]: > why mush add cast string here?

refactored

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 20 / 40 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|3|12|25.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|17|41.18%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|10|11|90.91%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 22 / 39 (56.41%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|17|41.18%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|5|11|45.45%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|10|11|90.91%|

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:495]: why must throw DdlException?

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:538]: remove empty line

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1364]: actually this a little confused to me, QualifiedNameContext named as db. but we get catalog from this db variable.
how about renaming the QualifiedNameContext variable name of ShowDatabasesStatementContext

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1385]: we do not need to check catalog isNull? just pass catalog, am i right? if it was true, we can simplify this code

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:495]: > why must throw DdlException?

cause this func calls listDbNames to get databases names which would throw DdlException
default List<String> listDbNames() throws DdlException {
     return Lists.newArrayList();
}

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 19 / 29 (65.52%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|5|11|45.45%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|10|11|90.91%|

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:495]: Why list db names throw Data **“Definition”** Language Exception, I think it's illegal

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:525]: @adzfolc Is it work for external databases?

Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:525]: yes, it works

Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:525]: > @adzfolc Is it work for external databases?

MySQL [demodb]> show databases from catalog_1;
+----------+
| Database |
+----------+
| default  |
| tn_test  |
| un_test  |
+----------+
3 rows in set (0.27 sec)

MySQL [demodb]> show databases from catalog_1 like 'default';
+----------+
| Database |
+----------+
| default  |
+----------+
1 row in set (0.00 sec)

MySQL [demodb]> show databases from catalog_1 like 'tn_test';
+----------+
| Database |
+----------+
| tn_test  |
+----------+
1 row in set (0.08 sec)

MySQL [demodb]> show databases from catalog_1 like '%test';
+----------+
| Database |
+----------+
| tn_test  |
| un_test  |
+----------+
2 rows in set (0.00 sec)

MySQL [demodb]> show databases from catalog_1 like 'aa';
Empty set (0.01 sec)

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java]: > 

refactored, now throws  AnalysisException, DdlException

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:495]: > 

it would be fixed by another pr, i need to wait for it got merged.

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java]: > remove empty line

done

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java]: > 

refactored.`qualifiedName` name now is changed to `dbName`.

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java]: > 

done

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:501]: what's the different between getClusterDbNames and localMetaStore.listDbNames @stephen-shelby

**Review by HangyuanLiu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:504]: todo: add metadataMgr to field in ShowExecutor

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:505]: 1、 add analyze in visitShowDatabases
2、metadataMgr.listDbNames;

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:501]: Both are currently the same. you could ref LocalMetastore.createDb() and unprotectCreateDb()

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:501]: There is now only one default cluster instance which named 'default_cluster'

Comment (caneGuy) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:501]: how about merge this 'if-else' block and just use connectorMetadata.get().listDbNames();?

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:501]: It is better to keep the previous usage, and then modify the corresponding call when the cluster is completely deleted.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 28 / 39 (71.79%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|5|8|62.50%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|8|12|66.67%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|10|11|90.91%|
|:large_blue_circle:|com/starrocks/common/ErrorCode.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 28 / 39 (71.79%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|5|8|62.50%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|8|12|66.67%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|10|11|90.91%|
|:large_blue_circle:|com/starrocks/common/ErrorCode.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java]: > It is better to keep the previous usage, and then modify the corresponding call when the cluster is completely deleted.

done.

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:501]: > you need check currentCatalog.

checked.first get current catalog from ctx, then check catalogName.

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java]: > catalogName

done

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java]: > todo: add metadataMgr as field in ShowExecutor

done

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
Comment (adzfolc) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java]: > 

done.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 30 / 39 (76.92%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|5|8|62.50%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|10|12|83.33%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|10|11|90.91%|
|:large_blue_circle:|com/starrocks/common/ErrorCode.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 29 / 38 (76.32%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|5|8|62.50%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|9|11|81.82%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|10|11|90.91%|
|:large_blue_circle:|com/starrocks/common/ErrorCode.java|1|1|100.00%|

**Review by stephen-shelby — APPROVED** [review]
LGTM

**Review by caneGuy — APPROVED** [review]

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by adzfolc** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 27 / 36 (75.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|4|6|66.67%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|8|10|80.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|11|13|84.62%|

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**
